packages/model-registry/upstream/frontend/src/app/pages/modelRegistry/screens/ModelVersions/ModelVersionsHeaderActions.tsx

```diff
@@ -35,28 +40,79 @@ const ModelVersionsHeaderActions: React.FC<ModelVersionsHeaderActionsProps> = ({
+          {latestModelVersion ? (
+            <DeployModalExtension
+              mv={latestModelVersion}
+              render={(buttonState, onOpenModal, isModalAvailable) => (
+                <Dropdown
+                  isOpen={isOpen}
+                  onSelect={() => setOpen(false)}
+                  onOpenChange={(open) => setOpen(open)}
+                  popperProps={{ position: 'end', appendTo: 'inline' }}
+                  toggle={(toggleRef) => (
+                    <MenuToggle
+                      variant="secondary"
+                      ref={toggleRef}
+                      onClick={() => setOpen(!isOpen)}
+                      isExpanded={isOpen}
+                      aria-label="Model action toggle"
+                      data-testid="model-action-toggle"
+                    >
+                      Actions
+                    </MenuToggle>
+                  )}
+                >
+                  <DropdownList>
+                    {isModalAvailable && (
+                      <>
+                        <DropdownItem
+                          isDisabled
+                          className="model-registry-section-header"
+                        >
+                          Latest version actions
+                        </DropdownItem>
+                        <DropdownItem
+                          onClick={() => {
+                            onOpenModal();
+                            setOpen(false);
+                          }}
+                          isAriaDisabled={!buttonState.enabled}
+                          tooltipProps={buttonState.tooltip ? { content: buttonState.tooltip } : undefined}
+                        >
+                          Deploy <strong>{latestModelVersion.name}</strong>
+                        </DropdownItem>
+                        <Divider />
+                      </>
+                    )}
+                    <ArchiveButtonDropdownItem setIsArchiveModalOpen={setIsArchiveModalOpen} />
+                  </DropdownList>
+                </Dropdown>
+              )}
+            />
+          ) : (
+            <Dropdown
+              isOpen={isOpen}
+              onSelect={() => setOpen(false)}
+              onOpenChange={(open) => setOpen(open)}
+              popperProps={{ position: 'end', appendTo: 'inline' }}
+              toggle={(toggleRef) => (
+                <MenuToggle
+                  variant="secondary"
+                  ref={toggleRef}
+                  onClick={() => setOpen(!isOpen)}
+                  isExpanded={isOpen}
+                  aria-label="Model action toggle"
+                  data-testid="model-action-toggle"
+                >
+                  Actions
+                </MenuToggle>
+              )}
+            >
+              <DropdownList>
+                <ArchiveButtonDropdownItem setIsArchiveModalOpen={setIsArchiveModalOpen} />
+              </DropdownList>
+            </Dropdown>
+          )}
```

Comment:
This seems like a large diff. It looks like the `<Dropdown>` props are duplicated exactly in both cases here and only its children are different depending on whether there is a version and the modal is available. Can we not just render the whole `<DeployModalExtension>` as a child of the DropdownList right before the ArchiveButtonDropdownItem and have it only render DropdownItems? or does that cause some loading problem?